### ISSUE TITLE
Import SMS not working for me

### ISSUE THREAD

**Issue by sorenkristensen80** [johanw666/Signal-Android#48]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I am unable to use the import feature (any of them), which is why I've installed it.
Import system SMS database jumps to white screen and back to start screen of the app.
Sometimes the (import finished - import of system database is finished) is in the drawer afterwards, but no chats/conversations are there.
Import plaintext backup says "No plaintext backup found". (I exportet my sms's with SMS Backup & Restore - and saved them to my phones storage).
Export - tried it just to check if that works, but no "Error, unable to write to storage".

### Steps to reproduce
- open import/export
- tap import system sms database
- tap import (if it's there)

- nothing happens

**Actual result:** Screen jumps back to start screen that is empty (no conversations/chats = no import)
**Expected result:** My sms backup should show as chats/conversations so I would have my previous conversations from Google Messages.

### Device info

**Device:** Huawei Mate 20 Pro
**Android version:** 10
**Emui version** 11.0.0
**Signal version:** 5.34.3.0-JW

### Link to debug log
https://debuglogs.org/android/5.34.3/2b41e8cbf96b09e995a9730c8d1a283157d347371be2906bb290f8ca17c26a7b

**Comment by QuendaltonBreed** [comment]
Same issue here. Soren, How did you install this version? Supposedly the one with import enabled?

**Comment by rambohenke** [comment]
Same problem here.

Also tried downloading "sms backup & restore" and made a backup with it.
Then tried the import "sms backup & restore" feature but just got "no backup found".

I'm kind of desperate getting my old messages into signal, what can I do to help debugging?

**Comment by johanw666** [comment]
Where did you put the backup file?

**Comment by rambohenke** [comment]
In the default location on the phone.

Den mån 16 maj 2022 15:46johanw666 ***@***.***> skrev:

**Comment by johanw666** [comment]
Which location is that for you? In Signal-JW the folder is 

- root of internal storage]/Signal
- Removable storage: [root of removable storage]/Android/data/org.thoughtcrime.securesms/files/Signal/PlaintextBackups

 Further you need to rename the file to SignalPlaintextBackup.xml

**Comment by johanw666** [comment]
Because I had to drop the sms import entirely due to changes in Signals internals I will close his as a won't fix.

**Comment by johanw666** [comment]
6.4.2.0-JW was the last version that supported direct SMS import from within Signal-JW. In newer versions you can do it indirectly by exporting the SMS database with SMS Backup and restore, and then import the resulting xml file in the plaintext import.

Note that both methods don't support MMS (although since 6.7 the export does export media messages in plaintext export, but only the text). Because MMS support was dropped by all telcos here years ago I testing it would be very difficulkt.

**Comment by twolaw** [comment]
Thanks.
And we can use this version as SMS app with no time limit ?

**Comment by johanw666** [comment]
The apk won't expre, like all my versions. However, it can be expected that at some point in time it will stop working due to changes in the server.

**Comment by twolaw** [comment]
6.4.2 doesn't work for me.
I had to install 6.12.3 and import SMS through Backup&Restore xml.

Which is the latest version not warning about "_SMS support ending_" in every conversation ?

**Comment by johanw666** [comment]
> Which is the latest version not warning about "SMS support ending" in every conversation ?

I don't know, I'd have to search in the code for that. Since that version will sooner or later stop working with the server anyway it doesn't really matter. Or do you want to use Signal only for sms?

**Comment by twolaw** [comment]
I found the SMS warning was added on Oct 13, 2022 with https://github.com/signalapp/Signal-Android/commit/b6db7e7af6b4cd744f9fb834051ce341d6edbcef 
So version 5.53 seems to be the last one without SMS warning. Too old version, I won't try.

> Or do you want to use Signal only for sms?

Yes. For the excellent UX of this app. If anyone would fork it to focus exclusively on the SMS features, that'd be great.
